sources/src/main/java/com/google/solutions/jitaccess/core/services/NotificationService.java

```diff
@@ -140,13 +135,45 @@ public void sendNotification(Notification notification) throws NotificationExcep
           notification.isReply()
             ? EnumSet.of(SmtpAdapter.Flags.REPLY)
             : EnumSet.of(SmtpAdapter.Flags.NONE));
-      }
-      catch (SmtpAdapter.MailException | AccessException | IOException e) {
+      } catch (SmtpAdapter.MailException | AccessException | IOException e) {
         throw new NotificationException("The notification could not be sent", e);
       }
     }
   }
 
+  public static class SlackNotificationService extends NotificationService {
+    private final SlackAdapter slackAdapter;
+
+    public SlackNotificationService(SlackAdapter slackAdapter) {
+      Preconditions.checkNotNull(slackAdapter);
+
+      this.slackAdapter = slackAdapter;
+    }
+
+    public boolean canSendNotifications() {
+      return true;
+    }
+
+    @Override
+    public void sendNotification(Notification notification) throws NotificationException {
+      Preconditions.checkNotNull(notification, "notification");
+
+      var message =
```

Comment:
Depending on the type of notification, the message should be different. Could we use the same template mechamism as in the SMTP adapter?